### ISSUE TITLE
[4.0] [a11y] Search form has no label

### ISSUE THREAD

**Issue by zwiastunsw** [joomla/joomla-cms#24468]
### Steps to reproduce the issue
1. Use screen reader, e.g. NVDA
2. Go to the search results list page (for ex. use search module to search for a popular word, e.g. joomla on a test site)
3. Go to serach field
or
Inspect the code 

![searchj4](https://user-images.githubusercontent.com/29527200/55524287-17bab400-568d-11e9-8038-d4544a82fb4b.png)

### Expected result
* The screen reader should announce a label for the search box

or
* The code should include a label for the field "Search" (label for="search-searchword"

### Actual result
Screen reader does not announce the name of the field
The field has no labels.

**Comment by brianteeman** [comment]
Is this on the front end?

If so is this Search or Smart Search?

**Comment by wilsonge** [comment]
https://github.com/joomla/joomla-cms/blob/4.0-dev/components/com_search/tmpl/search/default_form.php#L23

https://github.com/joomla/joomla-cms/blob/4.0-dev/components/com_finder/tmpl/search/default_form.php#L36

Both :)

**Comment by zwiastunsw** [comment]
On frontend. Only Search. In Smart Search label exist.

**Comment by brianteeman** [comment]
@wilsonge it is present for smart search but the markup is wrong
it says label for q where q is the name. 
it should say label for q where q is the id

I will work on both of these shortly on the train

**Comment by zwiastunsw** [comment]
> it says label for q where q is the name.
it should say label for q where q is the id

I agree!

**Comment by wilsonge** [comment]
I'm blind. Please carry on ;)

**Comment by zwiastunsw** [comment]
add id="q" into:
```	<input type="text" name="q" ```
The value of the `for` attribute must be the input  `ID` value, not `name`

**Comment by ghost** [comment]
closed as having Pull Request #24477
